pact/policy-manager/policy-manager.pact

```diff
@@ -1,103 +1,64 @@
-
 (namespace (read-msg 'ns))
 
 (module policy-manager GOVERNANCE
+
   (defcap GOVERNANCE ()
     (enforce-guard 'marmalade-admin ))
 
-  (use kip.token-policy-v2 [token-policies token-info concrete-policy NON_FUNGIBLE_POLICY QUOTE_POLICY ROYALTY_POLICY COLLECTION_POLICY GUARD_POLICY])
-
-  (defschema concrete-policy-list
-    policy-field:string
-    policy:module{kip.token-policy-v2}
-  )
-
-  (deftable concrete-policy-table:{concrete-policy-list})
-
-  (defconst CONCRETE_POLICY_V1_LIST
-    [NON_FUNGIBLE_POLICY QUOTE_POLICY ROYALTY_POLICY COLLECTION_POLICY GUARD_POLICY] )
+  ; (implements kip.token-policy-v2)
+  (use kip.token-policy-v2 [token-info concrete-policy NON_FUNGIBLE_POLICY QUOTE_POLICY ROYALTY_POLICY COLLECTION_POLICY GUARD_POLICY])
 
   ;; schema to save policy list in table
-  (defschema policies-list
-    concrete-policies:[module{kip.token-policy-v2}]
-    immutable-policies:[module{kip.token-policy-v2}]
-    adjustable-policies:[module{kip.token-policy-v2}]
+  (defschema concrete-policy-manager
```

Comment:
Won't it break the schema If we now want to add a concrete-policy?
An alternative might be to store them like this:

`(defschema concrete-policy
   policy:module{kip.token-policy-v2}
)
`
Then we could move the `concrete-policies` field from the ledger and have `(deftable concrete-policies:{concrete-policy})` instead. The key of the table would be the corresponding policy string that we have in kip.token-policy-v2

This also helps with the `get-concrete-policy` function further below. It now reads the complete object with 5 policies from the table everytime you want to get the module for a single policy. In this case you would just read a single entry per lookup (I don't know how much we would win with this, gas-wise)